src/SourceGenerators/Bit.SourceGenerators/AutoInject/AutoInjectNormalClassHandler.cs

```diff
@@ -33,14 +33,37 @@ namespace {classNamespace}
     [global::System.CodeDom.Compiler.GeneratedCode(""Bit.SourceGenerators"",""{AutoInjectHelper.GetPackageVersion()}"")]
     [global::System.Diagnostics.DebuggerNonUserCode]
     [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
-    public partial class {classSymbol.Name}
+    public partial class {GenerateClassName(classSymbol)}
     {{
         {GenerateConstructor(classSymbol, sortedMembers, baseEligibleMembers)}
     }}
 }}";
         return source;
     }
 
+    private static string GenerateClassName(INamedTypeSymbol classSymbol)
+    {
+        var genericTypesName = string.Empty;
+
+        if (classSymbol.IsGenericType)
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append("<");
+
+            foreach (var item in classSymbol.TypeParameters)
+            {
+                sb.Append($"{item.Name},");
+            }
+
+            sb.Length--;
+            sb.Append(">");
+
+            genericTypesName = sb.ToString();
+        }
+
+        return $"{classSymbol.Name}{genericTypesName}";
```

Comment:
please change this method's implementation based on a similar method in the `ComponentDetailsController` of the Bit.BlazorUI Playground (the GetTypeName method in the controller).